### PR TITLE
Release prep

### PR DESCRIPTION
**Pull request by NNcrawler**

### CODE REVIEW

File: version.txt [hunk 1]
```diff
@@ -1 +1 @@
-1.1.0
```
Comment (ravisuhag) [inline version.txt:1]: @NNcrawler Can we purge and then add this to clean up version history?

File: app/proc.go [hunk 2]
```diff
@@ -3,18 +3,37 @@ package app
...
+	"runtime"
 )
 
 //Run the server
 func Run() error {
 	ctx := context.Background()
 	ctx, cancel := context.WithCancel(ctx)
 
-	//@TODO - init config
+	registerCallback()
```
Comment (chakravarthyvp) [inline app/proc.go:15]: Could this be renamed this to `registerProcessMetrics`?

File: app/proc.go [hunk 3]
```diff
@@ -3,18 +3,37 @@ package app
...
 	//start server
 	StartServer(ctx, cancel)
 	logger.Info("App.Run --> Complete")
 	<-ctx.Done()
 	return nil
 }
+
+func registerCallback() {
+	m := &runtime.MemStats{}
+	metrics.RegisterCallback(func() {
```
Comment (chakravarthyvp) [inline app/proc.go:26]: explicitly calling of this as `callback` is confusing since the registered callback is not really an object that this client needs reference to in future. Would it make sense to call it likes of `registerMetricReporter` ? wdyt?

File: metrics/gaugeReporter.go [hunk 4]
```diff
@@ -0,0 +1,26 @@
+package metrics
+
```
Comment (chakravarthyvp) [inline metrics/gaugeReporter.go:2]: This is a nice refactoring.

However, the naming of this file could be better and not a gaugeReporter. Secondly it seems generic that it could be reused for any caller functions. If this file needs to deal with only metric reporters then it needs to be refactored accordingly.
For ex. This file simply registers a callback function and not necessarily validates that the registered function reports only metrics. In future I could add register any function (logging, closing connections, anything...) that gets called inside the goroutine.

File: metrics/gaugeReporter.go [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+import "time"
+
```
Comment (chakravarthyvp) [inline metrics/gaugeReporter.go:4]: Just thinking, could we move this refactoring out from this release and leave it in a separate PR? The reason: Since we are trying to release what's battle tested, we'd avoid any code changes for this release?
wdyt?

File: version.txt (hunk 1 above)
Comment (NNcrawler) [reply version.txt:1]: Okay, will purge this

File: metrics/gaugeReporter.go (hunk 5 above)
Comment (NNcrawler) [reply metrics/gaugeReporter.go:4]: Okay. Taking the feedbacks on another PR.

File: docs/concepts/structure.md [hunk 6]
```diff
@@ -1,2 +1,22 @@
 # Structure
```
Comment (chakravarthyvp) [inline docs/concepts/structure.md:1]: Nice

File: docs/contribute/release.md [hunk 7]
```diff
@@ -1,2 +1,8 @@
-# Release Process
+# Release
 
+Raccoon release tags follow [SEMVER](https://semver.org/) convention.
+
+Github workflow is used to build and push the built docker image to Docker hub.
+
+A release is triggered when a github tag of format `vM.m.p` is pushed. After release job is succeeded, a docker image of
```
Comment (chakravarthyvp) [inline docs/contribute/release.md:7]: 1. We should also be explicit that a release have to be accepted from the committers.
2. We should provide the command to push tags (say from which branch do we release)

3. The current flow pushes the version, latest as the tags: If built from the branch, it may override the released version?  For integration/branches, shall we use commit tags instead of x.y.z & latest docker tags?

File: docs/guides/monitoring.md [hunk 8]
```diff
@@ -3,6 +3,68 @@
...
+
+## Metrics Usages
+
+Following are key monitoring statistics that you can infer from Raccoon metrics. Refer to this section to get a clue of how to build alerting, dashboard, or do analysis on the metrics.
+### Data Loss
+To infer data loss, you can count [`kafka_messages_delivered_total`](https://odpf.gitbook.io/raccoon/reference/metrics#kafka_messages_delivered_total) with tag `success=false`. You can also infer loss rate by calculating 
+> count(`kafka_messages_delivered_total` `success=false`) / count(`kafka_messages_delivered_total`)
+### Latency
+Raccoon provides fine-grained metrics that denote latency. That gives clues as to where to look in case something goes wrong. In summary, these are key metrics for latency:
+- [`event_processing_duration_milliseconds`](https://odpf.gitbook.io/raccoon/reference/metrics#event_processing_duration_milliseconds) This metrics denotes overall latency. You need to look at other latency metrics to find the root cause when this metric is high.
```
Comment (chakravarthyvp) [inline docs/guides/monitoring.md:65]: need to clarify what each of the latency is (which time to which time). For ex., A worker latency computes the time taken between fetching the batch from the event channel and the each event acknowledged by kafka.

File: docs/guides/monitoring.md [hunk 9]
```diff
@@ -3,6 +3,68 @@
...
+- [`server_processing_latency_milliseconds`](https://odpf.gitbook.io/raccoon/reference/metrics#server_processing_latency_milliseconds) Correlate this metric with `event_processing_duration_milliseconds` to infer whether the issue is with Raccoon itself, or something wrong with the network, or the way [sent_time](https://github.com/odpf/proton/blob/main/odpf/raccoon/EventRequest.proto#L43) is generated.
+- [`worker_processing_duration_milliseconds`](https://odpf.gitbook.io/raccoon/reference/metrics#worker_processing_duration_milliseconds) High value of this metric indicates that the publisher is slow or can't keep up.
+
+### Backpressure
```
Comment (chakravarthyvp) [inline docs/guides/monitoring.md:69]: Could this go to troubleshooting section?

File: docs/guides/monitoring.md (hunk 8 above)
Comment (NNcrawler) [reply docs/guides/monitoring.md:65]: We have [an image](https://odpf.gitbook.io/raccoon/reference/metrics#event-delivery) to help the user understand when are these metrics are emitted. Will give a reference to this section be enough?

File: docs/contribute/release.md (hunk 7 above)
Comment (NNcrawler) [reply docs/contribute/release.md:7]: 1. Can you elaborate on this?
3. I think we should only release from master. Thus, the commit tag is not necessary.

File: docs/guides/monitoring.md (hunk 9 above)
Comment (NNcrawler) [reply docs/guides/monitoring.md:69]: Okay. Moving this 👍

File: docs/contribute/release.md (hunk 7 above)
Comment (chakravarthyvp) [reply docs/contribute/release.md:7]: agree, we can release only from master.

Comment (chakravarthyvp) [reply docs/contribute/release.md:7]: we can remove 2) since the committers take ownership of release.

File: docs/guides/monitoring.md (hunk 8 above)
Comment (chakravarthyvp) [reply docs/guides/monitoring.md:65]: My bad, didn't see the hyperlink, sounds good 👍
